x-pack/plugins/osquery/public/scheduled_query_groups/queries/query_flyout.tsx

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiCallOut,
+  EuiFlyout,
+  EuiTitle,
+  EuiSpacer,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiPortal,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { OsqueryManagerPackagePolicyConfigRecord } from '../../../common/types';
+import { CodeEditorField } from '../../queries/form/code_editor_field';
+import { Form, getUseField, Field } from '../../shared_imports';
+import { PlatformCheckBoxGroupField } from './platform_checkbox_group_field';
+import { ALL_OSQUERY_VERSIONS_OPTIONS } from './constants';
+import {
+  UseScheduledQueryGroupQueryFormProps,
+  useScheduledQueryGroupQueryForm,
+} from './use_scheduled_query_group_query_form';
+import { ManageIntegrationLink } from '../../components/manage_integration_link';
+
+const CommonUseField = getUseField({ component: Field });
+
+interface QueryFlyoutProps {
+  defaultValue?: UseScheduledQueryGroupQueryFormProps['defaultValue'] | undefined;
+  integrationPackageVersion?: string | undefined;
+  onSave: (payload: OsqueryManagerPackagePolicyConfigRecord) => Promise<void>;
+  onClose: () => void;
+}
+
+const QueryFlyoutComponent: React.FC<QueryFlyoutProps> = ({
+  defaultValue,
+  integrationPackageVersion,
+  onSave,
+  onClose,
+}) => {
+  const { form } = useScheduledQueryGroupQueryForm({
+    defaultValue,
+    handleSubmit: (payload, isValid) =>
+      new Promise((resolve) => {
+        if (isValid) {
+          onSave(payload);
```

Comment:
should `onClose` be called in the case this throws an exception?

Reply:
nope, we want to close the flyout only after the form was submitted properly